frontend/src/metabase/static-viz/components/XYChart/utils/scales.ts

```diff
@@ -101,13 +103,15 @@ const calculateYDomain = (
   const maxValue = max(values);
 
   return [
-    Math.min(0, minValue, goalValue ?? 0),
-    Math.max(0, maxValue, goalValue ?? 0),
+    Math.min(0, minValue, setMinValue ?? 0, goalValue ?? 0),
```

Comment:
I think this might be wrong. `Math.min` will return the minimum value from any of the parameters.

As I understand the `min` and `max` viz settings should either (not sure what's the behaviors are at the moment):
1. Don't allow the chart scale to go beyond these set values
2. set the chart scale to exactly these values (I don't think this is the case though)

If it's 1) then it's kind of like if the result from `Math.min(0, minValue, goalValue ?? 0)` is less than `setMinValue` uses `setMinValue` instead. Similarly, if `Math.max(0, maxValue, goalValue ?? 0)` is more than `setMaxValue` uses `setMaxValue` instead.

In either case, just making the `min` and `max` values one of the argument of `Math.min` and `Math.max` sounds wrong.

Reply:
You are definitely right. Sounds good matching this behavior with the app, it's just not what I expected initially. I thought we should let users set whatever they want for their charts, but seems we prevent them from shooting themselves in the foot in a lot of places too.